docs/operations/api-reference.md

```diff
@@ -614,6 +614,11 @@ Retrieve information about the segments of a task.
 
 Retrieve a [task completion report](../ingestion/tasks.md#task-reports) for a task. Only works for completed tasks.
 
+* `/druid/indexer/v1/lockedIntervals`
+
+Retrieve the list of Intervals locked by currently running ingestion/compaction tasks. The response contains a Map from
+Task IDs to the list of Intervals locked by the respective Tasks.
```

Comment:
why do we need task ids? can we just return a map of `datasource --> intervals`?

Reply:
In `CompactSegments.run()`, some currently running compaction tasks can get cancelled if their spec is out of date. We use the taskId to identify the intervals for these tasks getting cancelled so that we don't consider those intervals as locked.

This is the only case where taskId is useful. Otherwise, datasource would be sufficient.

Reply:
See here:
https://github.com/apache/druid/blob/5eb91aba52e22d9c72fd36f3994dc9b103ff5a69/server/src/main/java/org/apache/druid/server/coordinator/duty/CompactSegments.java#L147